### PR TITLE
tomcat: Add tomcat9 & persist & fixes download url

### LINKED ISSUE #283: [Request] Add Tomcat9
Tomcat 10 has been published in the `extras` bucket today. Are there any plans to publish Tomcat9 in `versions`?

Would it be possible to persist `conf` and `webapps` (see [Tomcat: conf and webapps directories are not persisted](https://github.com/lukesampson/scoop-extras/issues/6010))?

### PR DESCRIPTION
**Pull request by davidxuang**
* Closes #283 #68 #290
* Related: lukesampson/scoop-extras#5990

### CODE REVIEW

File: bucket/tomcat7.json [hunk 1]
```diff
@@ -16,7 +16,11 @@
     "env_set": {
         "CATALINA_HOME": "$dir",
         "CATALINA_BASE": "$dir"
-    },
+    },
```
Comment (davidxuang) [inline bucket/tomcat7.json:19]:
```suggestion
    },
```

File: bucket/tomcat8.json [hunk 2]
```diff
@@ -16,7 +16,11 @@
     "env_set": {
         "CATALINA_HOME": "$dir",
         "CATALINA_BASE": "$dir"
-    },
+    },
```
Comment (davidxuang) [inline bucket/tomcat8.json:19]:
```suggestion
    },
```

File: bucket/tomcat9.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            "url": "https://www.apache.org/dist/tomcat/tomcat-9/v9.0.45/bin/apache-tomcat-9.0.45-windows-x86.zip",
+            "hash": "sha512:9abe991fa8d06db0300a9f7a932ecc8374c369e088715382dc3a7a93a7c145f531125cb0a8d5653cf9b79cb99fbe7c3f3a93cbc19906cf3f323d123c31faec74"
+        }
+    },
+    "extract_dir": "apache-tomcat-9.0.45",
+    "bin": "bin\\catalina.bat",
+    "env_set": {
+        "CATALINA_HOME": "$dir",
+        "CATALINA_BASE": "$dir"
+    },
```
Comment (davidxuang) [inline bucket/tomcat9.json:24]:
```suggestion
    },
```

File: bucket/tomcat9.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    "persist": [
+        "conf",
+        "webapps"
+    ],
+    "checkver": {
+        "url": "https://www.apache.org/dist/tomcat/tomcat-9/?C=M;O=D",
```
Comment (janicmikes) [inline bucket/tomcat9.json:30]:
```suggestion
        "url": "https://downloads.apache.org/tomcat/tomcat-9/?C=M;O=D",
```

File: bucket/tomcat9.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        "regex": "v([\\d.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://www.apache.org/dist/tomcat/tomcat-$majorVersion/v$version/bin/apache-tomcat-$version-windows-x64.zip"
```
Comment (janicmikes) [inline bucket/tomcat9.json:36]:
```suggestion
                "url": "https://downloads.apache.org/tomcat/tomcat-$majorVersion/v$version/bin/apache-tomcat-$version-windows-x64.zip"
```

File: bucket/tomcat9.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            },
+            "32bit": {
+                "url": "https://www.apache.org/dist/tomcat/tomcat-$majorVersion/v$version/bin/apache-tomcat-$version-windows-x86.zip"
```
Comment (janicmikes) [inline bucket/tomcat9.json:39]:
```suggestion
                "url": "https://downloads.apache.org/tomcat/tomcat-$majorVersion/v$version/bin/apache-tomcat-$version-windows-x86.zip"
```

**Review by janicmikes — CHANGES_REQUESTED** [review]
I adjusted the url of tomcat-9 to match the redirects from apache.

You may want to adjust them for the 6,7 and 8 versions as well.

**Comment by HUMORCE** [comment]
@davidxuang hey, please update `tomcat8.json` and `tomcat9.json` to latest version.

**Review by HUMORCE — DISMISSED** [review]

**Review by HUMORCE — APPROVED** [review]

**Comment by HUMORCE** [comment]
sorry, i made a mistake here. 🤐

**Pull request merged**